libraries/AP_Mount/AP_Mount_STorM32_native.h

```diff
@@ -0,0 +1,138 @@
+
+    // init - performs any required initialisation for this instance
+    virtual void init(const AP_SerialManager& serial_manager);
+
+    // update mount position - should be called periodically
+    virtual void update();
+    virtual void update_fast();
+
+    // has_pan_control - returns true if this mount can control it's pan (required for multicopters)
+    virtual bool has_pan_control() const { return false; }
+
+    // set_mode - sets mount's mode
+    virtual void set_mode(enum MAV_MOUNT_MODE mode);
+
+    // status_msg - called to allow mounts to send their status to GCS using the MOUNT_STATUS message
+    virtual void status_msg(mavlink_channel_t chan);
+
+    // every mount should have this !!!
+    virtual bool is_armed(){ return _armed; }
+
+private:
+    // helper to handle corrupted rcin data
+    bool is_failsafe(void);
+
+    // interface to STorM32_lib
+    virtual size_t _serial_txspace(void);
+    virtual size_t _serial_write(const uint8_t *buffer, size_t size, uint8_t priority);
+    virtual uint32_t _serial_available(void);
+    virtual int16_t _serial_read(void);
+    uint16_t _rcin_read(uint8_t ch);
+
+    // internal variables
+    AP_HAL::UARTDriver *_uart;
+    AP_Mount::MountType _mount_type;
+    bool _initialised;              // true once the driver has been initialised
+    bool _armed;                    // true once the gimbal has reached normal operation state
+
+    enum TASKENUM {
+        TASK_SLOT0 = 0,
+        TASK_SLOT1,
+        TASK_SLOT2,
+        TASK_SLOT3,
+        TASK_SLOT4,
+        TASK_SLOTNUMBER,
+    };
+    uint64_t _task_time_last;
+    uint16_t _task_counter;
+
+    // we want to keep that info, the fields are 16 in size, so add one to convert it to string
+    //  _initialised = true indicates that these fields were set
+    char versionstr[16+1];
+    char namestr[16+1];
+    char boardstr[16+1];
+
+    // discovery functions
+    void find_gimbal_native(void);
+
+    // send info to gcs functions
+    bool _send_armeddisarmed;       // true when a armed/disarmed message should be send out
+    void send_text_to_gcs(void);
+
+    // bit mask, allows to enable/disable particular functions/features
+    enum BITMASKENUM {
+        SEND_STORM32LINK_V2 = 0x01,
+        SEND_CMD_SETINPUTS = 0x02,
+        GET_PWM_TARGET_FROM_RADIO = 0x04,
+        SEND_CMD_DOCAMERA = 0x08,
+    };
+    uint16_t _bitmask; //this mask is to control some functions
+
+    // storm32.Status in
+    struct {
+        float pitch_deg;
+        float roll_deg;
+        float yaw_deg;
+    } _status;
+    bool _status_updated;
+
+    void set_status_angles_deg(float pitch_deg, float roll_deg, float yaw_deg);
+    void get_status_angles_deg(float* pitch_deg, float* roll_deg, float* yaw_deg);
+
+    // target out
+    enum ANGLESTYPEENUM {
+        angles_deg = 0, //the STorM32 convention is angles in deg, not rad!
+        angles_pwm
```

Comment:
Can these enum values be changed to uppercase? They look like variables in code instead of enum values.